File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        boolean answer = true;
        
        Deque<String> deque = new ArrayDeque<>();
        
        for(int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == '(') {
                deque.addLast(Character.toString(s.charAt(i)));
            } else {
                if (deque.isEmpty()) {
                    return false;
                }
                deque.removeLast();
            }
        }

        answer = deque.isEmpty();        

        return answer;
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    x = 0
    for i in s:
        if (i == "("):
            x = x+1
        elif (i == ")"):
            x = x-1
            if x < 0:
                answer = False
    if x != 0:
        answer = False
    return answer